examples/infix_math_parser.py

```diff
@@ -0,0 +1,190 @@
+        exclude = f"{'|'.join([*symbols_variadic, *symbols_unary])}"
+        pattern = r"(?!\b(" + exclude + r")\b)(\b[a-zA-Z_][a-zA-Z0-9_]*\b)"
+        variable = Regex(pattern)
+
+        operands = variable | scientific | integer
+
+        # Forward declarations of variadric and unary function calls
+        variadic_call = Forward()
+        unary_call = Forward()
+
+        # The parsed expressions are assumed to follow a standard infix syntax. The operands
+        # of the infix syntax can be either the literal 'operands' defined above (these are singletons),
+        # or either a variadic function call or a unary function call. These latter two will be
+        # defined to be recursive.
+        #
+        # Note that the order of the operators in the second argument (the list) of infixNotation matters!
+        # The operation with the highest precedence is listed first.
+        infix_expn = infixNotation(
+            operands | variadic_call | unary_call,
+            [
+                (expop, 2, opAssoc.LEFT),
+                (signop, 1, opAssoc.RIGHT),
+                (multop, 2, opAssoc.LEFT),
+                (plusop, 2, opAssoc.LEFT),
+            ],
+        )
+
+        # These are recursive definitions of the forward declarations of the two type of function calls.
+        # In essence, the recursion continues until a singleton operand is encountered.
+        variadic_call <<= Group(
+            variadic_func_names + lparen + Group(delimitedList(infix_expn)) + rparen
+        )
+        unary_call <<= Group(unary_func_names + lparen + Group(infix_expn) + rparen)
+
+        self.expn = infix_expn
+
+        # The infix operations do not need to be in this list because they are handled by infixNotation() above.
+        # If new binary operations are to be added, they must be updated in the infixNotation() call (the list).
+        self.reserved_symbols: set[str] = symbols_unary | symbols_variadic
+
+        # It is assumed that the dicts in the three class variables have unique keys.
+        self.operator_mapping = {
+            **InfixExpressionParser.BINARY_OPERATORS,
+            **InfixExpressionParser.UNARY_OPERATORS,
+            **InfixExpressionParser.VARIADIC_OPERATORS,
+        }
+
+    def parse(self, str_expr: str) -> list:
+        """Parse a string expression into a list."""
+        return self.expn.parse_string(str_expr, parse_all=True)
+
+
+if __name__ == "__main__":
+    infix_parser = InfixExpressionParser()
+
+    expressions = [
+        "f_1 + f_2 - 1e-3",
+        "(x_1 + (x_2 * (c_1 + 3.3) / (x_3 - 2))) * 1.5",
+        "Max(Ln(x) + Lb(Abs(y)), Ceil(Sqrt(garlic) * 3), (potato ** 2) / 4, Abs(cosmic) + 10)",
+        "Max(Sqrt(Abs(x) + y ** 2), Lg(Max(cosmic, potato)), Ceil(Tanh(x) + Arctan(garlic)))",
+        "((garlic**3 - 2**Lb(cosmic)) + Ln(x**2 + 1)) / (Sqrt(Square(y) + LogOnePlus(potato + 3.1)))",
+    ]
+
+    for expr in expressions:
```

Comment:
Instead of doing your own loop (and handling exceptions), try `infix_parser.expn.run_tests(expressions)`

Reply:
I was not aware of this. Thanks!